frame/support/src/storage/generator/double_map.rs

```diff
@@ -378,12 +397,32 @@ impl<
 		}
 	}
 
+	fn iter_keys() -> Self::KeyIterator {
+		let prefix = G::prefix_hash();
+		Self::KeyIterator {
+			prefix: prefix.clone(),
+			previous_key: prefix,
+			drain: false,
+			closure: |raw_key_without_prefix| {
+				let mut k1_k2_material = G::Hasher1::reverse(raw_key_without_prefix);
+				let k1 = K1::decode(&mut k1_k2_material)?;
+				let mut k2_material = G::Hasher2::reverse(k1_k2_material);
+				let k2 = K2::decode(&mut k2_material)?;
+				Ok((k1, k2))
+			}
+		}
+	}
+
 	fn drain() -> Self::Iterator {
 		let mut iterator = Self::iter();
 		iterator.drain = true;
 		iterator
 	}
 
+	fn drain_keys() -> Self::KeyIterator {
```

Comment:
What behavioral difference is there between `drain()` and `drain_keys()`? I'd expect both to remove all keys and values, which implies that `drain_keys` is redundant.

Reply:
The only difference is that it doesn't decode the values. I'm ok with removing it since a draining iterator can be created from `iter_keys` just by calling `drain()` on the result anyway.